File: lista3/zadanie2/src/Square.cpp
#include <cmath>

#include "Square.hpp"

bool Square::CheckIfItCouldBe(ClientInput input)
{
    std::set<double> checkSides = readSidesFromInput(input);

    if(checkSides.size() > 1)
        return false;
    if(input.NextInput() != 90.)
        return false;
    return true;
}

Square::Square(ClientInput input) : Quadrangle(input)
{}

double Square::Area()
{
    return pow(*sides.begin(), 2.);
}
    
double Square::Perimeter()
{
    return (*sides.begin()) * 4.;
}

std::string Square::NameYourself()
{
    return "Kwadrat";
}

File: lista3/zadanie1/Square.java
import java.util.*;

class Square extends Quadrangle
{
    public static boolean CheckIfItCouldBe(ClientInput input)
    {
        Set<Double> checkSides = readSidesFromInput(input);

        if(checkSides.size() > 1)
            return false;
        if(input.NextInput() != 90.)
            return false;
        return true;
    }

    public Square(ClientInput input)
    {
        super(input);
    }

    public double Area()
    {
        return Math.pow(sides.iterator().next(), 2);
    }

    public double Perimeter()
    {
        return sides.iterator().next() * 4.;
    }

    public String NameYourself()
    {
        return "Kwadrat";
    }
};
